File: ViewModelExample/ViewModelExampleJava/app/src/main/java/com/tinlone/viewmodelexamplejava/HomeDataViewModel.java
package com.tinlone.viewmodelexamplejava;

import androidx.lifecycle.ViewModel;

public class HomeDataViewModel extends ViewModel {

    public int count = 0;

    private String defaultConfig = "default";

    public HomeDataViewModel() {
    }

    public HomeDataViewModel(String defaultConfig) {
        this.defaultConfig = defaultConfig;
    }

    public String readConfig(){
        return defaultConfig;
    }


    @Override
    protected void onCleared() {
        super.onCleared();
        // 做流的关闭，资源回收
    }
}

File: ViewModelExample/ViewModelExampleKotlin/app/src/main/java/com/tinlone/viewmodelexamplekotlin/HomeDataViewModel.kt
package com.tinlone.viewmodelexamplekotlin

import androidx.lifecycle.ViewModel

class HomeDataViewModel : ViewModel {
    var count = 0
    private var defaultConfig = "default"

    constructor() {}
    constructor(defaultConfig: String) {
        this.defaultConfig = defaultConfig
    }

    fun readConfig(): String {
        return defaultConfig
    }

    override fun onCleared() {
        super.onCleared()
        // 做流的关闭，资源回收
    }
}